src/cephadm/cephadm

```diff
@@ -219,7 +219,17 @@ class Monitoring(object):
     }  # type: ignore
 
 ##################################
-
+def populate_files(config_dir, config_files, uid, gid):
+    # type: (str, Dict, int, int) -> None
+    """create config files for different services"""
+    for fname in config_files:
+        config_file = os.path.join(config_dir, fname)
+        config_content = dict_get_join(config_files, fname)
+        logger.info('Write file: %s' % (config_file))
+        with open(config_file, 'w') as f:
+            os.fchown(f.fileno(), uid, gid)
+            os.fchmod(f.fileno(), 0o600)
+            f.write(config_content)
```

Comment:
can you please make this refactorization a seperate commit. I really want to avoid merging a single HUGE commit.

Reply:
@sebastian-philipp : We are in a hurry with this feature now.  I think that we are discussing this feature ( in different PRS)  for almost two months. In the first revision you pointed several important points that i think have been addressed.
This re-factorization was one of the first things you suggested us (not as a separate PR) . It would have been very easy to move this to another PR before Christmas, but in this moment,  this movement can carry us to more delays and to finish with this nice feature outside of our release.
We have tested this several times manually and passed the Integration tests... so I have to request  to do only the minimum changes needed.